analyser/waveform.py

```
@@ -79,134 +76,66 @@ def upload_waveform(file_path):
+        for store_track in store_tracks_to_process:
+            storeTrackId = store_track.get("id")
+            storeTrackUrl = store_track.get("url")
+            previewUrl = store_track.get("preview_url")
+            previewStartMs = store_track.get("start_ms")
+            previewEndMs = store_track.get("end_ms")
+            previewId = store_track.get("preview_id")
+            storeName = store_track.get("store_name")
+
+            print(f"Processing store track {storeTrackId} / {storeTrackUrl}")
+            try:
+                local_filename, _ = urllib.request.urlretrieve(previewUrl)
```

Ruff (0.15.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)